### ISSUE TITLE
Serialization of models

### ISSUE THREAD

**Issue by vondrejc** [sQUlearn/squlearn#275]
**Is your feature request related to a problem? Please describe.**
At the moment, there is no simple way for saving/loading of models. I have tried pickle, joblib, cloudpickle and none of them works.

**Describe the solution you'd like**
Enable the possibility of saving/loading of models with pickle or cloudpickle.

**Describe alternatives you've considered**
Alternatively, one could implement get_weights(), set_weights(weights: numpy.ndarray) for storing only the weights in the same way as in keras/tensorflow https://keras.io/guides/serialization_and_saving/

**Additional context**
Another solution is welcome!

**Comment by vondrejc** [comment]
It is possible to store models with dill https://pypi.org/project/dill/. Still, it would be nice to have a class method, which allows an official way how to store it without trying several packages.

**Comment by David-Kreplin** [comment]
Yes, indeed, we probably have to find our own way of storing the models and not solely rely on Dill. I think the biggest issue is the Executor class or the Backend class used within. We probably can't avoid reinitializing the Executor.
